packages/next-on-pages/src/buildApplication/generateFunctionsMap.ts

```diff
@@ -922,3 +956,22 @@ async function collectChunksConsumersInfosFromFuncDirectory(
 
 	return chunksConsumersInfos;
 }
+
+/**
+ * Copies bundled asset files into the __next-on-pages-dist__/assets folder as binary files (so that they can be
+ * fetched as binaries)
+ *
+ * @param distDir the __next-on-pages-dist__ directory's path
+ * @param bundledAssetsInfo map containing all the bundled assets files info collected during the build process
+ */
+async function copyBundledAssetFiles(
+	distDir: string,
+	bundledAssetsInfo: Map<string, BundledAssetInfo>,
+): Promise<void> {
+	const assetsDir = join(distDir, 'assets');
+	await mkdir(assetsDir);
+	for (const { filename, originalFileLocation } of bundledAssetsInfo.values()) {
+		const newLocation = join(assetsDir, `${filename}.bin`);
```

Comment:
This saddens me :(

We need to give frameworks a way to emit a manifest with content-types which Wrangler can pick up cc. @RamIdeas .